rpmlb/builder/base.py

```diff
@@ -78,15 +90,40 @@ def before(self, work, **kwargs):
     def after(self, work, **kwargs):
         pass
 
-    def prepare(self, package_dict):
+    def prepare(self, package_dict: Mapping[str, Any]):
+        """Prepare single package for a build.
+
+        Keyword arguments:
+            package_dict: A dictionary of package metadata.
+        """
+
         if 'name' not in package_dict:
             raise ValueError('package_dict is invalid.')
-        spec_file = '{0}.spec'.format(package_dict['name'])
-        if 'macros' in package_dict:
-            self.edit_spec_file_by_macros(spec_file, package_dict['macros'])
-        if 'replaced_macros' in package_dict:
-            self.edit_spec_file_by_replaced_macros(
-                spec_file, package_dict['replaced_macros'])
+
+        spec_file_path = Path('{name}.spec'.format_map(package_dict))
+
+        with self.edit_spec_file(spec_file_path) as (source_file, target_file):
+            content_stream = iter(source_file)  # Start modifications
```

Comment:
The `iter()` is not really needed here, but I need a new name for the iterator anyway (to allow for skipping of some edits), and I feel that `stream = iter(file)` is a bit better and more explicit than `stream = file`.

Counter-arguments welcome :)